modules/heat_conduction/include/userobjects/ViewFactorRayStudy.h

```diff
@@ -0,0 +1,166 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "RayTracingStudy.h"
+
+#include "RayTracingAngularQuadrature.h"
+
+/**
+ * RayTracingStudy used to generate Rays for view factor computation
+ * using the angular quadrature method.
+ */
+class ViewFactorRayStudy : public RayTracingStudy
+{
+public:
+  ViewFactorRayStudy(const InputParameters & parameters);
+
+  static InputParameters validParams();
+
+  void initialSetup() override;
+
+  /**
+   * Data structure used for storing all of the information needed to spawn
+   * Rays from a single element.
+   */
+  struct StartElem
+  {
+    StartElem() {}
+
+    StartElem(const Elem * elem,
+              const Elem * start_elem,
+              const unsigned short int incoming_side,
+              const BoundaryID bnd_id,
+              const std::vector<Point> & points,
+              const std::vector<Real> & weights)
+      : _elem(elem),
+        _start_elem(start_elem),
+        _incoming_side(incoming_side),
+        _bnd_id(bnd_id),
+        _points(points),
+        _weights(weights)
+    {
+      mooseAssert(_points.size() == _weights.size(), "Point and weight size not equal");
+    }
+
+    /// The element the points originate from
+    const Elem * _elem;
+    /// The element the trace will start from
+    const Elem * _start_elem;
```

Comment:
do you need elem once you have found start_elem?

Reply:
Yeah. It's used in generation for some normal flipping and so on: https://github.com/loganharbour/moose/blob/obstructed_vf/modules/heat_conduction/src/userobjects/ViewFactorRayStudy.C#L280